### PR TITLE
feature/improve error handling

### PR DESCRIPTION
**Pull request by tyriis**
:wave: @tarioch, I was forced to update the pre-commit-config aswel, as I was not able to load parts from gitlab. I have updated the pre-commit hook versions with `pre-commit autoupdate`. I hope this is fine for you.

This PR should solve `KeyErrors` caused by customization patches (not containing all meta information expected). #4 
The kustomization patches need to be excluded; it's worth hinting at this in the README.md.

```yaml
  - repo: https://github.com/tarioch/flux-check-hook
    rev: v0.2.0
    hooks:
      - id: check-flux-helm-values
        exclude: (cluster/apps/home/home-assistant/patches/db-init.yaml)
```

### CODE REVIEW

**Comment by tyriis** [comment]
added another commit to improve the readability of the helm errors and prevent `helm lint` if `helm pull` has failed

example `helm pull`
```
Check flux helm values......................................................Failed
- hook id: check-flux-helm-values
- exit code: 1

[ERROR] helm pull: 
Error: chart "csi-driver-nfs" version "v4.1.1" not found in https://raw.githubusercontent.com/kubernetes-csi/csi-driver-nfs/master/charts repository
```

example `helm lint`
```
Check flux helm values......................................................Failed
- hook id: check-flux-helm-values
- exit code: 1

[ERROR] helm lint: 
==> Linting csi-driver-nfs-v4.1.0.tgz
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/rbac-csi-nfs.yaml: unable to parse YAML: invalid Yaml document separator: kind: ClusterRole

Error: 1 chart(s) linted, 1 chart(s) failed
```

example kustomization patch
```
Check flux helm values......................................................Failed
- hook id: check-flux-helm-values
- exit code: 1

[ERROR] cluster/apps/home/home-assistant/patches/db-init.yaml: KeyError ('chart',)
```

File: pre_commit_flux/check_flux_helm_values.py [hunk 1]
```diff
@@ -66,8 +81,10 @@ def _validateFile(fileToValidate, repos):
                     stderr=subprocess.STDOUT,
                 )
                 if res.returncode != 0:
-                    print(res.stdout)
-                    exit(1)
+                    _collectErrors(
+                        {"source": "helm pull", "message": "\n{0}".format(res.stdout)}
```
Comment (tarioch) [inline pre_commit_flux/check_flux_helm_values.py]: you should be able to use a format string, something like ```f"\n{res.stdout}"```

File: pre_commit_flux/check_flux_helm_values.py [hunk 2]
```diff
@@ -78,8 +95,23 @@ def _validateFile(fileToValidate, repos):
                     stderr=subprocess.STDOUT,
                 )
                 if res.returncode != 0:
-                    print(res.stdout)
-                    exit(1)
+                    _collectErrors(
+                        {"source": "helm lint", "message": "\n{0}".format(res.stdout)}
```
Comment (tarioch) [inline pre_commit_flux/check_flux_helm_values.py]: format string as well

File: pre_commit_flux/check_flux_helm_values.py [hunk 3]
```diff
@@ -78,8 +95,23 @@ def _validateFile(fileToValidate, repos):
...
+
+
+def _collectErrors(error):
+    errors.append(error)
+
+
+def _printErrors():
+    for i in errors:
+        print(
+            Template("[ERROR] $source: $message").substitute(
```
Comment (tarioch) [inline pre_commit_flux/check_flux_helm_values.py]: I would also use a format string here

File: pre_commit_flux/check_flux_helm_values.py [hunk 4]
```diff
@@ -4,13 +4,28 @@
 import sys
 import tempfile
 from shlex import quote
+from string import Template
+
 import yaml
 
+errors: list = list()
```
Comment (tarioch) [inline pre_commit_flux/check_flux_helm_values.py]: errors: list = []

**Comment by tarioch** [comment]
Thank you very much for your contribution, I have added some small changes but otherwise this looks really nice.

File: pre_commit_flux/check_flux_helm_values.py (hunk 1 above)
Comment (tyriis) [reply pre_commit_flux/check_flux_helm_values.py]: there are to many pythonic ways i guess :trollface: will change.

File: pre_commit_flux/check_flux_helm_values.py (hunk 4 above)
Comment (tyriis) [reply pre_commit_flux/check_flux_helm_values.py]: refactored: https://github.com/tarioch/flux-check-hook/pull/5/commits/de5b285083cabd27f059eb1a9deade0f64f53854#diff-51d4385a76742bdc6b7dba6fbd4d3e84b2aaeb1f9b6b65b77e07a77d8a4dfc40R10

File: pre_commit_flux/check_flux_helm_values.py (hunk 3 above)
Comment (tyriis) [reply pre_commit_flux/check_flux_helm_values.py]: refactored: https://github.com/tarioch/flux-check-hook/pull/5/commits/de5b285083cabd27f059eb1a9deade0f64f53854#diff-51d4385a76742bdc6b7dba6fbd4d3e84b2aaeb1f9b6b65b77e07a77d8a4dfc40R103

File: pre_commit_flux/check_flux_helm_values.py (hunk 2 above)
Comment (tyriis) [reply pre_commit_flux/check_flux_helm_values.py]: refactored: https://github.com/tarioch/flux-check-hook/pull/5/commits/de5b285083cabd27f059eb1a9deade0f64f53854#diff-51d4385a76742bdc6b7dba6fbd4d3e84b2aaeb1f9b6b65b77e07a77d8a4dfc40R93

File: pre_commit_flux/check_flux_helm_values.py (hunk 1 above)
Comment (tyriis) [reply pre_commit_flux/check_flux_helm_values.py]: refactored: https://github.com/tarioch/flux-check-hook/pull/5/commits/de5b285083cabd27f059eb1a9deade0f64f53854#diff-51d4385a76742bdc6b7dba6fbd4d3e84b2aaeb1f9b6b65b77e07a77d8a4dfc40R79

**Comment by tyriis** [comment]
@tarioch thanks for your feedback, have changed to format string everywhere to have a consistent code style.

**Pull request merged**